src/commands/dev/lib/dev-router.ts

```diff
@@ -33,29 +52,27 @@ export default function(reqPath = '', routes?: RouteConfig[]): RouteResult {
 
       if (match) {
         let destPath: string = reqPathname;
+        let { headers } = routeConfig;
+
+        if (headers) {
+          Object.keys(headers).map((key)=>{
+            if (headers)
```

Comment:
@williamli You wrote `if (headers)` twice 😮

Reply:
Typescript thinks headers could be undefined in line 60.

Reply:
@styfle Typescript thinks headers could be undefined in line 60.

<img width="643" alt="Screenshot 2019-04-05 at 05 03 05" src="https://user-images.githubusercontent.com/179761/55588364-26be6680-5760-11e9-814d-1ed0dde7053b.png">
<img width="648" alt="Screenshot 2019-04-05 at 05 03 16" src="https://user-images.githubusercontent.com/179761/55588373-2aea8400-5760-11e9-8fde-cbdb3803a7d4.png">

Reply:
Strange. You could probably change line 55 to be `let { headers = {} } = routeConfig;` and then you wont need a `if` statement at all.

Reply:
@styfle Still doesn't like it
![image](https://user-images.githubusercontent.com/179761/55588515-8e74b180-5760-11e9-9afa-f7e3c5dbcd45.png)

Reply:
@styfle maybe this in line 555

```
let { headers } = {headers: {}, ...routeConfig};
```

Reply:
@williamli Try this:

```ts
let headers = routeConfig.headers || {};
```

Reply:
@styfle nar, back to immutability problem

Reply:
https://github.com/zeit/now-cli/pull/2098